src/metpy/io/_tools.py

```diff
@@ -138,7 +138,7 @@ class Enum:
     def __init__(self, *args, **kwargs):
         """Initialize the mapping."""
         # Assign values for args in order starting at 0
-        self.val_map = {ind: a for ind, a in enumerate(args)}
```

Comment:
You're right, this could be better written as:
```python
self.val_map = dict(enumerate(args))
```
Unfortunately as this PR stands, this only removes the line--and is thus incorrect.